enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+
+## Proposal
+
+- Set the `InfrastructureTopology` to have the same value as the
+  `ControlPlaneTopology` when the number of workers is 1, instead of setting its
+  value to `SingleReplica` as it's done today.
+
+- Set masters as schedulable when workers are less or equal to 1. Today,
+  masters are set to schedulable only when there are no workers available.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to be able to install a 4-nodes
+  cluster as a single installation step so that my cluster will be able to
+  handle workloads that require replication.
+
+
+### API Extensions
+
+- Sets the `infrastructureTopology` field to the value of`ControlPlaneTopology` in the
+  `config.openshift.io/v1/infrastructure` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+- Sets the `mastersSchedulable` field to true in the
+  `config.openshift.io/v1/scheduler` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This should not introduce newer risks other than the existing ones in a compact
+cluster deployment.
+
+## Design Details
+
+### Open Questions [optional]
+
+Nothing at the moment
+
+### Test Plan
+
+
+- Add unit tests to ensure the Infrastructure and Scheduler settings are properly set for the different combinations
+
+- Add a periodic job (nightly?) job to run OpenShift tests on a 4-node cluster
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+This enhancement does not have an impact on existing clusters as it's intended
+to affect only day0 operations.
+
+### Version Skew Strategy
+
+Does not apply
+
+### Operational Aspects of API Extensions
+
+
+- Allows for a 4-nodes, highly available, cluster to be deployed as a day0
+  task. It reduces the time required for such deployments to be created.
+
+- Does not impact general deployment functionalities in a negative way.
+
+- Allowing workloads on the master nodes may affect the master's health if the
+  traffic and workloads are too high.
+
+#### Failure Modes
+
+Same as a compact cluster
+
+#### Support Procedures
+
+- The same support procedures used for a compact cluster should be used for a
+  4-nodes cluster. I would appreciate help filling in this section with more
+  details.
+
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+A 4-nodes cluster (a.k.a 3+1) is generally not a recommended architecture. By
```

Comment:
If this isn't a recommended configuration, we should have some more justification for supporting it explicitly than "users asked for it." Can you add more details about the use case(s) above?

Reply:
I will rephrase this because I don't think saying "it is generally not recommended" is fair here. IT's not recommended for specific cases due to the motivations I mentioned in this paragraph. I do think it can be ok for other scenarios. 

I will add more details